// ebay-api/lib/api/restful/buy/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order = exports.Offer = exports.MarketplaceInsights = exports.Marketing = exports.Feed = exports.Deal = exports.Browse = void 0;
const index_js_1 = __importDefault(require("./browse/index.js"));
exports.Browse = index_js_1.default;
const index_js_2 = __importDefault(require("./feed/index.js"));
exports.Feed = index_js_2.default;
const index_js_3 = __importDefault(require("./marketing/index.js"));
exports.Marketing = index_js_3.default;
const index_js_4 = __importDefault(require("./offer/index.js"));
exports.Offer = index_js_4.default;
const index_js_5 = __importDefault(require("./order/index.js"));
exports.Order = index_js_5.default;
const index_js_6 = __importDefault(require("./deal/index.js"));
exports.Deal = index_js_6.default;
const index_js_7 = __importDefault(require("./marketplaceInsights/index.js"));
exports.MarketplaceInsights = index_js_7.default;
