// ebay-api/lib/api/apiFactory.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const index_js_1 = __importDefault(require("./index.js"));
const index_js_2 = require("./restful/buy/index.js");
const index_js_3 = require("./restful/commerce/index.js");
const index_js_4 = require("./restful/developer/index.js");
const index_js_5 = require("./restful/postOrder/index.js");
const index_js_6 = require("./restful/sell/index.js");
const index_js_7 = __importDefault(require("./traditional/index.js"));
class ApiFactory extends index_js_1.default {
    constructor() {
        super(...arguments);
        this._restful = {};
    }
    createBuyApi() {
        return {
            browse: this.createRestfulApi(index_js_2.Browse),
            feed: this.createRestfulApi(index_js_2.Feed),
            marketing: this.createRestfulApi(index_js_2.Marketing),
            offer: this.createRestfulApi(index_js_2.Offer),
            order: this.createRestfulApi(index_js_2.Order),
            deal: this.createRestfulApi(index_js_2.Deal),
            marketplaceInsights: this.createRestfulApi(index_js_2.MarketplaceInsights),
        };
    }
    createCommerceApi() {
        return {
            catalog: this.createRestfulApi(index_js_3.Catalog),
            charity: this.createRestfulApi(index_js_3.Charity),
            identity: this.createRestfulApi(index_js_3.Identity),
            notification: this.createRestfulApi(index_js_3.Notification),
            media: this.createRestfulApi(index_js_3.Media),
            translation: this.createRestfulApi(index_js_3.Translation),
            taxonomy: this.createRestfulApi(index_js_3.Taxonomy),
        };
    }
    createDeveloperApi() {
        return {
            analytics: this.createRestfulApi(index_js_4.Analytics),
            keyManagement: this.createRestfulApi(index_js_4.KeyManagement),
        };
    }
    createPostOrderApi() {
        return {
            cancellation: this.createRestfulApi(index_js_5.Cancellation),
            case: this.createRestfulApi(index_js_5.Case),
            inquiry: this.createRestfulApi(index_js_5.Inquiry),
            return: this.createRestfulApi(index_js_5.Return),
        };
    }
    createSellApi() {
        return {
            account: this.createRestfulApi(index_js_6.Account),
            analytics: this.createRestfulApi(index_js_6.Analytics),
            compliance: this.createRestfulApi(index_js_6.Compliance),
            fulfillment: this.createRestfulApi(index_js_6.Fulfillment),
            inventory: this.createRestfulApi(index_js_6.Inventory),
            marketing: this.createRestfulApi(index_js_6.Marketing),
            metadata: this.createRestfulApi(index_js_6.Metadata),
            recommendation: this.createRestfulApi(index_js_6.Recommendation),
            finances: this.createRestfulApi(index_js_6.Finances),
            feed: this.createRestfulApi(index_js_6.Feed),
            logistics: this.createRestfulApi(index_js_6.Logistics),
            negotiation: this.createRestfulApi(index_js_6.Negotiation),
            listing: this.createRestfulApi(index_js_6.Listing),
        };
    }
    get traditional() {
        if (this._traditional) {
            return this._traditional;
        }
        return (this._traditional = new index_js_7.default(this.config, this.req, this.auth));
    }
    createTradingApi() {
        return this.traditional.createTradingApi();
    }
    createShoppingApi() {
        return this.traditional.createShoppingApi();
    }
    createFindingApi() {
        return this.traditional.createFindingApi();
    }
    createClientAlertsApi() {
        return this.traditional.createClientAlertsApi();
    }
    createMerchandisingApi() {
        return this.traditional.createMerchandisingApi();
    }
    createRestfulApi(RestfulApiClass) {
        const id = RestfulApiClass.id;
        return (this._restful[id] || (this._restful[id] = new RestfulApiClass(this.config, this.req, this.auth)));
    }
}
exports.default = ApiFactory;
