// ebay-api/lib/api/restful/commerce/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Media = exports.Translation = exports.Taxonomy = exports.Notification = exports.Identity = exports.Charity = exports.Catalog = void 0;
const index_js_1 = __importDefault(require("./catalog/index.js"));
exports.Catalog = index_js_1.default;
const index_js_2 = __importDefault(require("./charity/index.js"));
exports.Charity = index_js_2.default;
const index_js_3 = __importDefault(require("./identity/index.js"));
exports.Identity = index_js_3.default;
const index_js_4 = __importDefault(require("./media/index.js"));
exports.Media = index_js_4.default;
const index_js_5 = __importDefault(require("./notification/index.js"));
exports.Notification = index_js_5.default;
const index_js_6 = __importDefault(require("./taxonomy/index.js"));
exports.Taxonomy = index_js_6.default;
const index_js_7 = __importDefault(require("./translation/index.js"));
exports.Translation = index_js_7.default;
